modules/aws_ecs_fargate/main.tf

```diff
@@ -0,0 +1,247 @@
+terraform {
+  required_providers {
+    aws = {
+        source = "hashicorp/aws"
+        version = "~> 4.0"
+    }
+  }
+}
+
+provider "aws" {
+  region = var.aws_region
+}
+
+resource "aws_ecs_cluster" "this" {
+  name = "${var.deployment_name}-ecs"
+
+  setting {
+    name  = "containerInsights"
+    value = var.ecs_insights_enabled
+  }
+}
+
+resource "aws_cloudwatch_log_group" "this" {
+  name              = "${var.deployment_name}-ecs-log-group"
+  retention_in_days = var.log_retention_in_days
+}
+
+resource "aws_db_instance" "this" {
+  identifier                    = "${var.deployment_name}-rds-instance"
+  allocated_storage            = 80
+  instance_class               = var.rds_instance_class
+  engine                       = "postgres"
+  engine_version               = "13.7"
+  db_name                      = "hammerhead_production"
+  username                     = aws_secretsmanager_secret_version.rds_username.secret_string
+  password                     = aws_secretsmanager_secret_version.rds_password.secret_string
+  port                         = 5432
+  publicly_accessible          = var.rds_publicly_accessible
+  vpc_security_group_ids       = [aws_security_group.rds.id]
+  performance_insights_enabled = var.rds_performance_insights_enabled
+  
+  skip_final_snapshot          = true
+  apply_immediately           = true
+}
+
+resource "aws_ecs_service" "retool" {
+  name                               = "${var.deployment_name}-main-service"
+  cluster                            = aws_ecs_cluster.this.id
+  task_definition                    = aws_ecs_task_definition.retool.arn
+  desired_count                      = var.ecs_service_count
+  deployment_maximum_percent         = var.maximum_percent
+  deployment_minimum_healthy_percent = var.minimum_healthy_percent
+  launch_type                        = "FARGATE"
+
+  load_balancer {
+    target_group_arn = aws_lb_target_group.this.arn
+    container_name   = "retool"
+    container_port   = 3000
+  }
+
+  network_configuration {
+    subnets         = var.subnet_ids
+    security_groups = [aws_security_group.alb.id]
+  }
+}
+
+resource "aws_eip" "retool_ip" {
+  vpc = true
+}
+
+resource "aws_nat_gateway" "retool_nat_gateway" {
```

Comment:
so we expect the customer to come with a public subnet w/ an internet gateway inside of it? I wonder if we should just do this for them?

I recommend creating a networking.tf file that sets this stuff up if these variables are not provided. don't we also need route table rules to make sure traffic from our private subnet actually makes it to this NAT gateway?

i think this is a good example https://dev.betterdoc.org/infrastructure/2020/02/04/setting-up-a-nat-gateway-on-aws-using-terraform.html